### PR TITLE
Fix flake8 warnings for new flake8

### PR DESCRIPTION
**Pull request by illicitonion**

### CODE REVIEW

File: synapse/app/__init__.py [hunk 1]
```diff
@@ -12,3 +12,23 @@
...
 # limitations under the License.
+
+import sys
+sys.dont_write_bytecode = True
+
+from synapse.python_dependencies import (
+    check_requirements, MissingRequirementError
+)  # NOQA
+
+if __name__ == '__main__':
```
Comment (erikjohnston) [inline synapse/app/__init__.py]: Does this actually work still when you do `python -m synapse.app.homeserver`?

**Comment by erikjohnston** [comment]
Eurgh, can't we turn off the option that forces us to put `and` and `or`'s on the previous line?

File: synapse/app/__init__.py (hunk 1 above)
Comment (illicitonion) [inline synapse/app/__init__.py]: Bleugh, not it doesn't, I checked that the file gets loaded but not the actual functionality.

How about we just remove the __name__ check? Either way this code won't work without its deps...

Comment (erikjohnston) [inline synapse/app/__init__.py]: *defers to @NegativeMjark*

**Comment by illicitonion** [comment]
Disabled that check. Fixed a bunch more that it now finds.

File: setup.cfg [hunk 2]
```diff
@@ -1,4 +1,4 @@
-[build_sphinx]
+[[build_sphinx]
```
Comment (erikjohnston) [inline setup.cfg]: Is this on purpose?

Comment (illicitonion) [inline setup.cfg]: Nope. Undone.

**Comment by erikjohnston** [comment]
LGTM

**Comment by erikjohnston** [comment]
Assuming tests pass

File: synapse/app/__init__.py (hunk 1 above)
Comment (NegativeMjark) [inline synapse/app/__init__.py]: It seems rude to sys.exit() unless you being run as the main program, but I don't think it should cause any problems for any of the likely ways this code will be run.

File: setup.cfg [hunk 3]
```diff
@@ -16,3 +16,4 @@ ignore =
 
 [flake8]
 max-line-length = 90
+ignore = W503
```
Comment (NegativeMjark) [inline setup.cfg]: Could you add a comment explaining what ``W503`` is and why we are disabling it

**Pull request merged**